arcgis-ios-sdk-samples/Route & Directions/Route around barriers/RouteParametersViewController.swift

```diff
@@ -56,13 +49,26 @@ class RouteParametersViewController: UITableViewController {
         switch sender {
         case findBestSequenceSwitch:
             routeParameters?.findBestSequence = sender.isOn
+            if sender.isOn {
+                tableView?.insertSections([1], with: .fade)
+            } else {
+                tableView?.deleteSections([1], with: .fade)
+            }
             setupUI()
         case preserveFirstStopSwitch:
-            routeParameters?.preserveFirstStop = preserveFirstStopSwitch?.isOn == true
+            routeParameters?.preserveFirstStop = sender.isOn
         case preserveLastStopSwitch:
-            routeParameters?.preserveLastStop = preserveLastStopSwitch?.isOn == true
+            routeParameters?.preserveLastStop = sender.isOn
         default:
             break
         }
     }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        if routeParameters?.findBestSequence == false {
+            return 1
+        }
+        return super.numberOfSections(in: tableView)
```

Comment:
I would suggest always getting `super`'s value and then subtracting off the section you don't want. Something like this:

```swift
override func numberOfSections(in tableView: UITableView) -> Int {
    let superCount = super.numberOfSections(in: tableView)
    if routeParameters?.findBestSequence ?? false {
        return superCount - 1
    } else {
        return superCount
    }
}
```